### ISSUE TITLE
coins pre-validated meta store

### ISSUE THREAD

**Issue by jwtlake** [MRN-Code/penny-collector#27]
In the case of Dicom files there could be upwards of 1000 individual files that would all generate the same meta information. After the first coins meta upload/validation we don't need to do any additional api calls. 

We should implement some kind of client side cacheing to store recent penny objects (meta data) and have penny collector check the cache before making validations calls.

**Comment by swashcap** [comment]
Hm. We’ll need some way of confirming we’ve received data on a file-by-file basis. If the network goes down, the API errors out, or the daemon :boom: explodes we’ll need to resume upload intelligently.

**Comment by drewlandis** [comment]
@jwtlake agreed.  For client-side, we had initially thought something like redis.  More recently, we've been leaning towards just something that is local to the Node memory.  No need for an external service to maintain that info.  The "buffer" should be one file deep (e.g. only ever store metadata for one file.  As soon as a new one comes in, we dump the old one).

@swashcap agreed.  We do need to track data that we have received on a file-by-file basis.  I think this may be somewhat of a separate issue.  Maybe I'm misunderstanding you tho?

This would be good to talk about at the Design Review today.  I think that discussing in-person will be easier than via github issues.
